p4-16/spec/P4-16-spec.mdk

```diff
@@ -7058,11 +7058,17 @@ diagnostics.
 
 # Annotations { #sec-annotations }
 
-Annotations are similar to C# attributes and Java annotations. They
-are a simple mechanism for extending the P4 language to some limited
-degree without changing the grammar. To some degree they subsume C `#pragma`s.
-Annotations are attached to types, fields, variables, etc. using the `@`
-syntax (as shown explicitly in the P4 grammar), and have an optional body:
+Annotations are similar to C# attributes and Java annotations. They are a simple
+mechanism for extending the P4 language to some limited degree without changing
+the grammar. To some degree they subsume C `#pragma`s. Annotations are attached
+to types, fields, variables, etc. using the `@` syntax (as shown explicitly in
+the P4 grammar). Unstructured annotations, or just "annotations," have an
+optional body; structured annotations have a mandatory body, containing at least
+a pair of square brackets `[]`.
+
+Structured annotations and  unstructured annotations must not use the same
+`name`. Thus, a given `name` can only be applied to one type of annotation or
+the other.
```

Comment:
This sounds like it might forbid the following, but it's not clear whether that is the intent. We should clarify and ensure that the implementation matches.
```
@my_anno(1) table T { ... }
@my_anno[2] table U { ... }
```